opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -220,42 +220,97 @@ message Span {
+    DEPRECATED_STATUS_CODE_UNKNOWN_ERROR       = 2;
+    DEPRECATED_STATUS_CODE_INVALID_ARGUMENT    = 3;
+    DEPRECATED_STATUS_CODE_DEADLINE_EXCEEDED   = 4;
+    DEPRECATED_STATUS_CODE_NOT_FOUND           = 5;
+    DEPRECATED_STATUS_CODE_ALREADY_EXISTS      = 6;
+    DEPRECATED_STATUS_CODE_PERMISSION_DENIED   = 7;
+    DEPRECATED_STATUS_CODE_RESOURCE_EXHAUSTED  = 8;
+    DEPRECATED_STATUS_CODE_FAILED_PRECONDITION = 9;
+    DEPRECATED_STATUS_CODE_ABORTED             = 10;
+    DEPRECATED_STATUS_CODE_OUT_OF_RANGE        = 11;
+    DEPRECATED_STATUS_CODE_UNIMPLEMENTED       = 12;
+    DEPRECATED_STATUS_CODE_INTERNAL_ERROR      = 13;
+    DEPRECATED_STATUS_CODE_UNAVAILABLE         = 14;
+    DEPRECATED_STATUS_CODE_DATA_LOSS           = 15;
+    DEPRECATED_STATUS_CODE_UNAUTHENTICATED     = 16;
   };
 
-  // The status code. This is optional field. It is safe to assume 0 (OK)
-  // when not set.
-  StatusCode code = 1;
+  // The deprecated status code. This is an optional field.
+  //
+  // This field is deprecated and is replaced by `code` field below. See backward
+  // compatibility notes below. According to our stability guarantees this field
+  // will be removed in 12 months, on Sep 26, 2021. All usage of old senders and
+  // receivers that do not understand the `code` field MUST be phased out by then.
+  DeprecatedStatusCode deprecated_code = 1 [deprecated=true];
 
   // A developer-facing human readable error message.
   string message = 2;
+
+  // For the semantics of status codes see
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#statuscanonicalcode
+  enum StatusCanonicalCode {
+    // The default status.
+    STATUS_CANONICAL_CODE_UNSET               = 0;
+    // The Span has been validated by an Application developers or Operator to have
+    // completed successfully.
+    STATUS_CANONICAL_CODE_OK                  = 1;
+    // The Span contains an error.
+    STATUS_CANONICAL_CODE_ERROR               = 2;
+  };
+
+  // The status code.
+  StatusCanonicalCode code = 3;
+
+  // IMPORTANT: Backward compatibility notes:
+  //
+  // To ensure any two pair of senders and receivers continues to correctly signal and
+  // interpret erroneous situation the senders and receivers SHOULD follow these rules:
+  //
+  // 1. Old senders and receives that are not aware of `code` field will continue using
+  // `deprecated_code` field to signal and interpret erroneous situation.
+  //
+  // 2. New senders, which are aware of `code` field should set both `deprecated_code`
+  // and `code` fields according to the following rules:
+  //
+  //   if code==STATUS_CANONICAL_CODE_UNSET then `deprecated_code` must be
+  //   set to DEPRECATED_STATUS_CODE_OK.
+  //
+  //   if code==STATUS_CANONICAL_CODE_ERROR then `deprecated_code` must be
+  //   set to DEPRECATED_STATUS_CODE_UNKNOWN_ERROR.
+  //
+  //   if code==STATUS_CANONICAL_CODE_OK then `deprecated_code` must be
```

Comment:
```suggestion
  //   if code==STATUS_CANONICAL_CODE_OK then `deprecated_code` MUST be
```

You could move that before the error rule to match the sorting order of the definition above.

Reply:
Done, thanks.